src/Containers/Microsoft.NET.Build.Containers/ContainerHelpers.cs

```diff
@@ -279,7 +280,13 @@ internal static bool IsValidEnvironmentVariable(string envVar)
                 throw new ArgumentException(Resources.Resource.GetString(nameof(Strings.InvalidImageName)));
             }
             var loweredImageName = containerImageName.ToLowerInvariant();
-            normalizedImageName = imageNameCharacters.Replace(loweredImageName, "-");
+            var potentialNormalizedName = imageNameCharacters.Replace(loweredImageName, "-");
+            if (potentialNormalizedName.All(c => c == '-'))
```

Comment:
should we also check if first char is not a dash? error message says "The first character of the image name must be a lowercase letter or a digit"

Reply:
I think this check is covered by 278-281 above - what do you think?

Reply:
regex on line 286 might replace the first character to a dash - will it be an issue?

Reply:
I don't think the regex will replace the first character ever, because the check on 278 (`!Char.IsLetterOrDigit`) is stronger than the regex, so if that check fails we should never hit the regex case, and if that check passes it should also pass the regex.

Reply:
`Char.IsLetterOrDigit` supports any Unicode character, whereas regex has only latin letters.

So
```
var imageName = "㓳";
var isLetter = Char.IsLetterOrDigit(imageName[0]); //true
var potentialNormalizedName = imageNameCharacters.Replace(imageName, "-"); //"-"
```

Reply:
You're exactly correct - so I should move the Char.IsLetterOrDigit check to a Latin check regex instead and then what I _thought_ was happening would actually happen. great catch.

Reply:
Using similar check `(c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')` may be more lightweight.